api/transaction.go

```diff
@@ -101,19 +101,127 @@ func (a *API) signTxHandler(w http.ResponseWriter, r *http.Request) {
 					}
 				}
 			}
+		case *models.Tx_NewProcess:
+			txNewProcess := tx.GetNewProcess()
+			// check the tx fields
+			if txNewProcess == nil || txNewProcess.Process == nil || txNewProcess.Nonce == 0 {
+				ErrInvalidTxFormat.With("missing fields").Write(w)
+				return
+			}
+			// check the tx subtype
+			switch txNewProcess.Txtype {
+			case models.TxType_NEW_PROCESS:
+				// generate a new faucet package if it's not present and include it in the tx
+				if txNewProcess.FaucetPackage == nil {
+					faucetPkg, err := a.account.FaucetPackage(organizationSigner.AddressString(), bootStrapFaucetAmount)
+					if err != nil {
+						ErrCouldNotCreateFaucetPackage.WithErr(err).Write(w)
+						return
+					}
+					txNewProcess.FaucetPackage = faucetPkg
+					tx = &models.Tx{
+						Payload: &models.Tx_NewProcess{
+							NewProcess: txNewProcess,
+						},
+					}
+				}
+			}
 		case *models.Tx_SetProcess:
-			log.Infow("signing SetProcess transaction", "user", user.Email)
+			txSetProcess := tx.GetSetProcess()
+			// check the tx fields
+			if txSetProcess == nil || txSetProcess.ProcessId == nil {
+				ErrInvalidTxFormat.With("missing fields").Write(w)
+				return
+			}
+			// check the tx subtype
+			switch txSetProcess.Txtype {
+			case models.TxType_SET_PROCESS_STATUS:
+				// check if the process status is in the tx
+				if txSetProcess.Status == nil {
+					ErrInvalidTxFormat.With("missing status field").Write(w)
+					return
+				}
+			case models.TxType_SET_PROCESS_CENSUS:
+				// check if the process census is in the tx
+				if (txSetProcess.CensusRoot == nil || txSetProcess.CensusURI == nil) && txSetProcess.CensusSize == nil {
+					ErrInvalidTxFormat.With("missing census fields").Write(w)
+					return
+				}
+			case models.TxType_SET_PROCESS_QUESTION_INDEX:
+				// check if the process question index is in the tx
+				if txSetProcess.QuestionIndex == nil {
+					ErrInvalidTxFormat.With("missing question index field").Write(w)
+					return
+				}
+			case models.TxType_SET_PROCESS_RESULTS:
+				// check if the process results are in the tx
+				if txSetProcess.Results == nil {
+					ErrInvalidTxFormat.With("missing results field").Write(w)
+					return
+				}
+			case models.TxType_SET_PROCESS_DURATION:
+				// check if the process duration is in the tx
+				if txSetProcess.Duration == nil {
+					ErrInvalidTxFormat.With("missing duration field").Write(w)
+					return
+				}
+			}
+			// include the faucet package in the tx if it's not present
+			if txSetProcess.FaucetPackage == nil {
+				faucetPkg, err := a.account.FaucetPackage(organizationSigner.AddressString(), bootStrapFaucetAmount)
```

Comment:
Same here, you need to set the exact price amount for the tx. In the case of these transactions, you should get the price from the vocdoni-node API.